### PR TITLE
Extract code to build custom properties defs to script and use in prebuild-css and webpack

### PR DESCRIPTION
**Pull request by jsnajdr**
This PR ensures that the `public/custom-properties.css` file is generated every time we need it.

The code that generates it was extracted to a `bin/` script. That script is called from `npm run prebuild-css`.

The second call site is now a webpack plugin that taps into the pre-compilation hook. That ensures that the file is re-generated on every webpack run.

**How to test:**
Verify that the build proceeds correctly in all possible scenarios, especially SDK and CI-related.

### CODE REVIEW

**Comment by sirreal** [comment]
If this fixes the underlying issue, care to revert #30019 here?

**Comment by jsnajdr** [comment]
> If this fixes the underlying issue, care to revert #30019 here?

Did you mean #30211?

File: webpack.config.js [hunk 1]
```diff
@@ -355,6 +367,7 @@ function getWebpackConfig( { cssFilename, externalizeWordPressPackages = false }
 					},
 				} ),
 			shouldEmitStats && new webpack.ProgressPlugin( createProgressHandler() ),
+			new BuildCustomPropertiesCssPlugin(),
```
Comment (blowery) [inline webpack.config.js]: nice, good call to bake it into the webpack pipeline

File: webpack.config.js [hunk 2]
```diff
@@ -41,6 +42,17 @@ const shouldCheckForCycles = process.env.CHECK_CYCLES === 'true';
 const codeSplit = config.isEnabled( 'code-splitting' );
 const isCalypsoClient = process.env.CALYPSO_CLIENT === 'true';
 
+/**
+ * Plugin that generates the `public/custom-properties.css` file before compilation
+ */
+class BuildCustomPropertiesCssPlugin {
+	apply( compiler ) {
+		compiler.hooks.compile.tap( 'BuildCustomPropertiesCssPlugin', () =>
+			execSync( 'node bin/build-custom-properties-css.js' )
```
Comment (blowery) [inline webpack.config.js]: are we guaranteed on the current working dir here with the sdk?

**Comment by jsnajdr** [comment]
@sirreal #30211 reverted.

File: webpack.config.js (hunk 2 above)
Comment (jsnajdr) [reply webpack.config.js]: > are we guaranteed on the current working dir here with the sdk?

Not sure. I don't even know what scenarios to test for that.

Comment (jsnajdr) [reply webpack.config.js]: I just experimented a bit... and building from a different cwd doesn't work even on `master`. Topic for another PR, as it's a frequent request to make it work.

**Pull request merged**